Source/WebCore/Modules/url-pattern/URLPatternCanonical.cpp

```diff
@@ -36,10 +36,10 @@ namespace WebCore {
 
 static constexpr auto dummyURLCharacters { "https://www.webkit.org"_s };
 
-static bool isInvalidIPv6HostCodePoint(auto codepoint)
+static bool isValidIPv6HostCodePoint(auto codepoint)
 {
     static constexpr std::array validSpecialCodepoints { '[', ']', ':' };
-    return !isASCIIHexDigit(codepoint) && std::find(validSpecialCodepoints.begin(), validSpecialCodepoints.end(), codepoint) != validSpecialCodepoints.end();
+    return isASCIIHexDigit(codepoint) || std::find(validSpecialCodepoints.begin(), validSpecialCodepoints.end(), codepoint) != validSpecialCodepoints.end();
```

Comment:
Boolean logic always confuses me and I don't quite understand why nothing has to be changed about the `std::find`, but the tests suggest this works fine.

Reply:
I think it's okay as-is. I guess what I didn't grok is that the negation applied to both conditions.